core/trino-main/src/main/java/io/trino/operator/project/DictionaryAwarePageFilter.java

```diff
@@ -119,25 +119,15 @@ public SelectedPositions filter(ConnectorSession session, Page page)
 
     private static SelectedPositions selectDictionaryPositions(DictionaryBlock dictionaryBlock, boolean[] selectedDictionaryPositions)
     {
+        int[] positions = new int[dictionaryBlock.getPositionCount()];
```

Comment:
this might potentially end up as indexes within `DictionaryBlock` (e.g in we could use `DictionaryBlock` directly in `InputPageProjection`). However, I don't think it does yet